adapter/authzOpa/authzOpa.go

```diff
@@ -0,0 +1,164 @@
+	"istio.io/mixer/pkg/status"
+	"istio.io/mixer/template/authz"
+)
+
+type (
+	builder struct {
+		adapterConfig *config.Params
+	}
+
+	handler struct {
+		policy      string
+		checkMethod string
+
+		context  context.Context
+		compiler *ast.Compiler
+
+		env adapter.Env
+	}
+)
+
+///////////////// Configuration Methods ///////////////
+
+func (b *builder) SetAuthzTypes(map[string]*authz.Type) {}
+
+func (b *builder) SetAdapterConfig(cfg adapter.Config) {
+	b.adapterConfig = cfg.(*config.Params)
+}
+
+func (b *builder) Validate() (ce *adapter.ConfigErrors) {
+	if len(b.adapterConfig.CheckMethod) == 0 {
+		ce = ce.Appendf(GetInfo().Name,
+			"CheckMethod was not configured")
+	}
+
+	parsed, err := ast.ParseModule("", string(b.adapterConfig.Policy))
+	if err != nil {
+		ce = ce.Appendf(GetInfo().Name,
+			"Failed to parse the OPA policy: %v", err)
+	}
+
+	compiler := ast.NewCompiler()
+	compiler.Compile(map[string]*ast.Module{"": parsed})
+	if compiler.Failed() {
+		ce = ce.Appendf(GetInfo().Name,
+			"Failed to compile the OPA policy: %v", compiler.Errors)
+	}
+
+	return
+}
+
+func (b *builder) Build(context context.Context, env adapter.Env) (adapter.Handler, error) {
+	h := handler{
+		policy:      b.adapterConfig.Policy,
+		checkMethod: b.adapterConfig.CheckMethod,
+		env:         env,
+	}
+
+	parsed, _ := ast.ParseModule("", string(h.policy))
+
+	compiler := ast.NewCompiler()
+	compiler.Compile(map[string]*ast.Module{"": parsed})
+
+	h.compiler = compiler
+	h.context = context
+
+	return h, nil
+}
+
+////////////////// Runtime Methods //////////////////////////
+
+func (h handler) HandleAuthz(context context.Context, instance *authz.Instance) (adapter.CheckResult, error) {
+	variables := make(map[string]interface{})
+
+	for k, v := range instance.Subject {
+		variables[k] = v
+	}
+	for k, v := range instance.Resource {
+		variables[k] = v
+	}
+	for k, v := range instance.Verb {
+		variables[k] = v
+	}
+
+	rego := rego.New(
+		rego.Compiler(h.compiler),
+		rego.Query(h.checkMethod),
+		rego.Input(variables),
+	)
+
+	// Run evaluation.
+	rs, err := rego.Eval(h.context)
+	if err != nil {
+		return adapter.CheckResult{
+			Status: status.WithPermissionDenied(fmt.Sprintf("authzOpa: request was rejected: %v", err)),
+		}, nil
+	}
+
+	if len(rs) != 1 {
+		return adapter.CheckResult{
+			Status: status.WithPermissionDenied("authzOpa: request was rejected"),
+		}, nil
+	}
+
+	result, ok := rs[0].Expressions[0].Value.(bool)
+	if !ok || result == false {
+		return adapter.CheckResult{
+			Status: status.WithPermissionDenied("authzOpa: request was rejected"),
+		}, nil
+	}
+
+	return adapter.CheckResult{
+		Status: rpc.Status{Code: int32(rpc.OK)},
+	}, nil
+}
+
+func (h handler) Close() error {
+	return nil
+}
+
+////////////////// Bootstrap //////////////////////////
+
+// GetBuilderInfo returns the BuilderInfo associated with
+// this adapter implementation.
+func GetInfo() adapter.Info {
+	return adapter.Info{
+		Name:        "authzOpa",
```

Comment:
Name must be all lowercase, alphanumeric, starting with a letter. That's why the presubmit is failing for you.

In general though I think we should just call this adapter "opa", or even "openpolicyagent"; we don't typically include the template name in the adapter name. @mandarjog @douglas-reid thoughts w.r.t. naming?

Reply:
Changed name to opa